src/chunk.c

```diff
@@ -1002,21 +1002,21 @@ verify_used_chunk(uint16_t region, uint16_t offset)
  * \return true if the region is valid.
  */
 static int
-region_is_valid(uint16_t region)
+region_is_valid(uint32_t region)
 {
   int result;
   Region *rp;
   RegionHeader *rhp;
-  int used_count;
-  int free_count;
-  int free_bytes;
-  int largest_free;
+  uint32_t used_count;
+  uint32_t free_count;
+  uint32_t free_bytes;
+  uint32_t largest_free;
   unsigned int total_derefs;
-  int len;
+  uint32_t len;
   int was_free;
   int dump;
-  uint16_t next_free;
-  uint16_t offset;
+  uint32_t next_free;
+  uint32_t offset;
```

Comment:
Fixes warnings. They were already comparisons to uint16_t before. Which were safe against ints. But now they're properly safe.